x/tally/keeper/payout.go

```diff
@@ -0,0 +1,60 @@
+package keeper
+
+import (
+	"sort"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/sedaprotocol/seda-chain/x/tally/types"
+)
+
+// CalculateCommitterPayouts constructs distribution messages that
+// pay the fixed gas cost for each commiter of a given data request.
+func (k Keeper) CalculateCommitterPayouts(ctx sdk.Context, req types.Request) (types.DistributionMessages, error) {
+	result := types.DistributionMessages{
+		Messages:   []types.DistributionMessage{},
+		RefundType: types.DistributionTypeTimedOut,
+	}
+	if len(req.Commits) == 0 {
+		return result, nil
+	}
+
+	gasCost, err := k.GetGasCostCommitment(ctx)
+	if err != nil {
+		return types.DistributionMessages{}, err
+	}
+
+	i := 0
+	committers := make([]string, len(req.Commits))
+	for k := range req.Commits {
+		committers[i] = k
+		i++
+	}
+	sort.Strings(committers)
+
+	distMsgs := make([]types.DistributionMessage, len(committers))
+	for i, committer := range committers {
+		distMsgs[i] = types.DistributionMessage{
+			Kind: types.DistributionKind{
+				ExecutorReward: &types.DistributionExecutorReward{
+					Identity: committer,
+					Amount:   math.NewIntFromUint64(gasCost),
```

Comment:
We should take the predetermined `gas_cost_commitment` for the commitment and multiply it with the `gas_price` as it was specified in the data request.

And probably we need to separate calculating the gas consumed for all the actors and the actual payout. In theory it's possible to create a DR with a really low `exec_gas_limit` and have stupid executors trying to commit to it (as they could know up front that it's not worth executing). Right now we would end up with a scenario where the distribution messages tally up to more than what was attached to the DR.

Reply:
To complete this PR I multiplied the commit gas cost with the request gas price ([e70a899](https://github.com/sedaprotocol/seda-chain/pull/461/commits/e70a89964bbab3a2ac7c4107cd4e664f17d9d57c))
But @Thomasvdam and I discussed that module should return the gas consumption amounts without multiplying them with the gas price. The contract can then multiply the reported gas amounts with the gas price included in the data request when it processes the payments. The contract would also make sure to pay the data providers and executors in alphabetical order until it runs out of funds. Cc: @gluax

Reply:
Currently the contract seems to be missing distribution message types for slashing executors and paying data providers. These should also be supported through distribution messages, right?

Reply:
Hmmm, I suppose those are fine to calculate on the contract side since it's a sudo message and no therefore has no gas fee.

However, how is alphabetical order determined for the data providers and executors? Is it by address alphabetical order? How do burn messages fit into that? We'd have to iterate over the list do that on the contract side... once to sort once to do the payouts. I feel like it might be easier to pass the data pre-sorted to the contracts.

I thought we weren't slashing executors, just paying the best actor more? As for paying a data provider you need a new enum on the message type, not a new type of distribution message yea?

Reply:
Ah sorry, slashing is not a priority at the moment. You're right, we would need a new enum (or `DistributionKind` right?) for paying a data provider. I think the order can be arbitrary, but I think paying data providers -> paying executors -> burn tally gas would be good? 

As for sorting, you're correct actually. The module will return the distribution messages sorted by their addresses since it works with sorted lists of reveals and proxy keys.

Reply:
Ok yea that makes sense to me.

Reply:
Probably good to get more inputs on the ordering. I think there’s an argument to be made to have the burn as the highest priority, as this effectively raises the value of the SEDA that the other actors are already holding or will earn in the future. :p